src/Uno.UI/UI/Xaml/Media/Imaging/BitmapImage.cs

```diff
@@ -128,19 +128,37 @@ internal BitmapImage(string stringSource) : base(stringSource)
 
 		public BitmapImage() { }
 
-		private void RaiseDownloadProgress(DownloadProgressEventArgs args)
+		private void RaiseDownloadProgress(int progress = 0)
 		{
-			DownloadProgress?.Invoke(this, args);
+			if (DownloadProgress is { } evt)
```

Comment:
Is there some functional advantage of this pattern over `Foo?.Invoke()`? It's noticeably more verbose.

Reply:
Nope.  But the goal here was to prevent the creation of a `DownloadProgressEventArgs` when there's no handlers on the event.

Note: this event wasn't implemented yet on Uno. I implemented it for Wasm only for now.

Reply:
Don't worry, if you call `DownloadProgress?.Invoke(this, new Args())` and there are no handlers, no Args object will be created :)

Reply:
Also: you kept the `?.Invoke` so there is double `null` check

Reply:
@davidjohnoliver you're right ! https://dotnetfiddle.net/lhGZ8Y

But that's not the case here because it was created by the caller and pass to this method. The refactoring was to pass only the progress instead of an event args.